### PR TITLE
Add details to strip charge description.

### PR DESCRIPTION
**Pull request by umgelurgel**
Fix discount coupons valid for individual users.

### CODE REVIEW

File: assopy/tests/factories/order.py [hunk 1]
```diff
@@ -11,3 +17,25 @@ class Meta:
 
     payment = 'cc' # cc because stripe is a credit card
     items = []
+
+
+class CouponFactory(factory.django.DjangoModelFactory):
+    class Meta:
+        model = 'assopy.Coupon'
+
+    conference = factory.Iterator(conference.models.Conference.objects.all())
```
Comment (artcz) [inline assopy/tests/factories/order.py:26]: isn't that executed at load/run time (and we probably want it to execute it on "call time" [whatever proper terminology is here. :P])

File: conference/cart.py [hunk 2]
```diff
@@ -158,6 +157,7 @@ def cart_step4_payment(request, order_uuid):
             token=request.POST.get("stripeToken"),
             token_type=request.POST.get("stripeTokenType"),
             email=request.POST.get("stripeEmail"),
+            description=f"payment for order {order.pk} by {request.user.email}",
```
Comment (artcz) [inline conference/cart.py:160]: i think adding order.code here might also make sense.

File: conference/orders.py [hunk 3]
```diff
@@ -134,7 +134,7 @@ def calculate_order_price_including_discount(
             end_validity__gte=timezone.now().date(),
         )
 
-        if not coupon.valid(user=for_user):
+        if not coupon.valid(user=for_user.assopy_user):
```
Comment (artcz) [inline conference/orders.py:137]: :+1:

File: tests/test_cart.py [hunk 4]
```diff
@@ -105,6 +108,35 @@ def test_can_buy_tickets_if_fare_is_available(db, user_client):
     assert Ticket.objects.all().count() == 10
 
 
+def test_can_apply_personal_ticket_coupon(db, user_client):
```
Comment (artcz) [inline tests/test_cart.py:111]: :pray:

**Review by artcz — APPROVED** [review]
LGTM with some comments.

File: assopy/tests/factories/order.py (hunk 1 above)
Comment (umgelurgel) [reply assopy/tests/factories/order.py:26]: I copied it from another factory, it should probably just fetch the current conference instead :) Updating.

**Pull request merged**